certora/spec/timelockAuthorizerSasha.spec

```diff
@@ -16,23 +16,23 @@ invariant matchingGenralActionIds()
 
 // STATUS - verified
 // all _delaysPerActionId are less or equal than MAX_DELAY
-invariant notGreaterThanMax(env e, bytes32 actionId)
+invariant notGreaterThanMax( bytes32 actionId)
     _delaysPerActionId(actionId) <= MAX_DELAY()
     {
-        preserved setDelay(bytes32 actionId1, uint256 delay) with (env e2) {
+        preserved setDelay(bytes32 actionId1, uint256 delay) with (env e2) { //e2 needed? 
             require actionId == actionId1;
-            require delay <= MAX_DELAY();
+            require delay <= MAX_DELAY(); // is this needed - the code does not enforce this? 
         }
     }
 
 
 // STATUS - verified
 // Any executableAt from _scheduledExecutions is not far more in the future than MAX_DELAY
-invariant notFarFuture(env e, uint256 actionIndex)
-    getSchedExeExecutableAt(actionIndex) <= e.block.timestamp + MAX_DELAY()
+invariant notFarFuture(uint timestamp , uint256 actionIndex) // I prefer just timestamp since e is not used.
```

Comment:
why?

Reply:
I think it is because it is more readable. When you pass `e` it is not obvious, that you only use `e.block.timestamp` in the code. When you pass `timestamp`, anybody can tell, that you only need the `timestamp` in the body by only looking at the function parameters.
Is there any other reason?